### PR TITLE
Add parsnp

### PR DESCRIPTION
**Pull request by Cashalow**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by Cashalow** [comment]
The osX build fails because clang is not compatible with openMP for conda-forge version apparently ? I have removed the osX build because I'm not sure about how to fix this.

File: recipes/parsnp/build.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+./autogen.sh
+./configure --prefix=${PREFIX}
+make
+sed -i.bak '1i#!/usr/bin/env python\n' Parsnp.py
+sed -i.bak 's/sys\.stderr\.write(\ BOLDME/sys\.stdout\.write(\ BOLDME/' Parsnp.py
```
Comment (mbargull) [inline recipes/parsnp/build.sh]: Can you give any information as to why changing the program's behavior this way is necessary?

Comment (Cashalow) [reply recipes/parsnp/build.sh]: For the second sed: whenever the program used to be called, the version and the link to the documentation would be sent to stderr, prompting any test to fail for instance.

For the first one: I thought this was standard procedure for directing the system to use python for interpreting the script files.

Comment (mbargull) [reply recipes/parsnp/build.sh]: But your change will alter the program's output (on `stdout`). I don't believe this should be done.
The `stderr` stream is commonly used not only for outputting errors but also logging information. Hence I believe the original usage is totally legitimate. If this caused any tests to fail, then those tests are likely broken

Comment (Cashalow) [reply recipes/parsnp/build.sh]: Sorry you are right I confused to things at the same time. It was calling parsnp without any arguments that throws an error code, I'll remove the second sed.

Comment (mbargull) [reply recipes/parsnp/build.sh]: Great, thanks!

**Comment by Cashalow** [comment]
Please don't merge this yet. I am still working on it and something is not quite right. Is it okay if I add to the recipe a few test files (one genome as gbk and one as fasta ) so that I can check that parsnp is running as expected ?

**Comment by mbargull** [comment]
> Is it okay if I add to the recipe a few test files

If the build tests are *very lightweight* (both, in terms of runtime as well as storage), that's fine.
No matter if you add the test data or not, I'd like to refer to
http://bioconda.github.io/contribute-a-recipe.html#test-locally
for the early stages of creating a recipe. Testing locally has the benefit that one can inspect the created Conda packages from the local build cache (i.e., `conda-bld` folder of the Miniconda installation) manually or even install it with `conda [create|install] -n test-env --use-local package-name`.

**Comment by bgruening** [comment]
@Cashalow any help needed?

**Comment by Cashalow** [comment]
Thanks a lot for asking @bgruening !

I had more or less given up on adding parsnp on bioconda because the version that is shipped in the github, that I tried compiling, is actually different from the binaries made available on the website ( http://harvest.readthedocs.io/en/latest/content/parsnp/quickstart.html)

Basically I ran into the exact same problem as mentioned on this thread when somebody tried to add it to homebrew: 

https://github.com/marbl/parsnp/issues/3

Given that the software as not been updated for almost 3 years now, that the last time the author answered to an issue on the github was more than one year ago, I basically gave up understanding how to build it correctly.

The alternative would be to add directly the binary available on the website to bioconda but I see little added value for that... What do you think ?

**Comment by bgruening** [comment]
If it is not needed lets close this recipe.
It is ok to add binaries to bioconda, but we prefer to compile from source for several reasons. I guess we can make an exception here if you need this package. Your call.

Just to understand it correctly, your test in this PR is not sufficient and it actually is broken what you currently have?

**Comment by Cashalow** [comment]
The binary that I obtain in this PR is working, the only problem is that it does not correspond at all to the binary distributed : it is not called the same way, needs a config file, and there is no information as to how to use this binary. 

I understood from the homebrew-related thread that I linked that the final distributed binary was compiled using pyinstaller but I did not managed. 

As it stands, the binary I obtain is useless.

**Comment by Cashalow** [comment]
Closing now. I got the software to work more or less as intended but the compile binary now throws Segmentation errors and I don't really know why. I'm not sure I see the point of adding the frozen binary so we will leave it like that for the time being.

**Pull request closed without merging**